src/types/object.js

```diff
@@ -2,22 +2,19 @@ import g from "../bindings/mod.js";
 import { getName } from "../utils/string.ts";
 import { handleCallable, handleStructCallable } from "./callable.js";
 import { objectByGType } from "../utils/gobject.js";
-import { GConnectFlags } from "../bindings/enums.js";
-import { createCallback } from "./callback.js";
 import { handleSignal } from "./signal.js";
 import { handleProp } from "./prop.js";
+import { GType } from "../bindings/enums.js";
 
-function extendObject(target, info) {
+function getParentClass(info) {
   const parent = g.object_info.get_parent(info);
 
   if (parent) {
     const gType = g.registered_type_info.get_g_type(parent);
 
     const ParentClass = objectByGType(gType);
-    Object.setPrototypeOf(target.prototype, ParentClass.prototype);
-    //Object.assign(target.__signals__, ParentClass.__signals__);
 
-    g.base_info.unref(parent);
```

Comment:
is there a reason to keep the parent reference?

Reply:
Yes, afaict if the reference is dropped, it won't be possible to construct the child types. Although, I'm unsure about this and some testing may be required. I can test it after #25 is reviewed, as that depends on it.

Reply:
Did you test this?

Reply:
I just tested this, and it seems it's indeed uneeded.